File: 0118-pascals-triangle/0118-pascals-triangle.java
class Solution {
    public List<List<Integer>> generate(int numRows) {
        List<List<Integer>> result = new ArrayList<>();

        for(int i = 0; i<numRows; i++){
            result.add(new ArrayList<Integer>());
            result.get(i).add(1);

            for(int j = 1; j<i; j++){
                result.get(i).add(result.get(i-1).get(j-1)+result.get(i-1).get(j));
            }
            if(i>0)
                result.get(i).add(1);
        }

        return result;
    }
}

File: 0118-pascals-triangle/0118-pascals-triangle.py
class Solution:
    def generate(self, numRows: int) -> List[List[int]]:
        pascals = [[1], [1,1]]

        if numRows < 3:
            return pascals[:numRows]
        
        for i in range(1, numRows-1):
            pascals.append([1])
            for j in range(len(pascals[i])-1):
                pascals[i+1].append(pascals[i][j]+pascals[i][j+1])
            pascals[i+1].append(1)

        return pascals